### Joystick Filter settings
debug: true
use_medianfilter: false
use_kalmannfilter: true
adaptive_kalman: true
display_variance_matrices: false
publish_frequency: 10.0

## joystick mapping parameters
# kv and kw are the linear constants mapping the joystick input (normalized) to linear and angular velocity v and w. 
# Since joystick input is normalized, kv and kw also indicate the maximum velocity output.
kv: 1.0
kw: 4.0
deadzone: 0.25 # v_threshold = kv*deadzone
control_exponent: 1.5

## Median filter parameters
buffer_size: 5

## Kalman filter parameters
P0: 0.1
Hx: 5.0
Hy: 5.0
alpha: 0.8 # learning rate or alternatively forget factor for adaptive Kalman Higher means longer memory. Between 0 and 1. 

## The following parameters are initial values. They can be adaptively estimated by the Adaptive Kalman Filter by setting adaptive_kalman to true.
cov_x: 0.1
cov_y: 0.1
cov_vx: 0.1
cov_vy: 0.1
Qx: 0.1
Qy: 0.1

